levels.go

```diff
@@ -626,6 +643,35 @@ func (lc *levelsController) fillTablesL0(cd *compactDef) bool {
 	return true
 }
 
+const minSkippedTableSize = 1024 * 1024
+
+func (lc *levelsController) fillBottomTables(cd *compactDef, overlappingTables []*table.Table) {
+	for _, t := range overlappingTables {
+		// If none of the top tables contains the range in an overlapping bottom table,
+		// we can skip it during compaction to reduce write amplification.
+		var added bool
+		for _, topTbl := range cd.top {
+			iter := topTbl.NewIteratorNoRef(false)
+			iter.Seek(t.Smallest())
+			if iter.Valid() && y.CompareKeysWithVer(iter.Key(), t.Biggest()) <= 0 {
+				cd.bot = append(cd.bot, t)
+				added = true
+				break
```

Comment:
return here and remove the the added boolean flag check.

Reply:
Only one bottom table will be added if we return here.
We need to add more bottom tables.